homeassistant/components/intellifire/__init__.py

```diff
@@ -42,9 +47,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     finally:
         await ift_control.close()
 
+    # Extract API Key and User_ID from ift_control
+    # Eventually this will migrate to using IntellifireAPICloud
+
+    if CONF_USER_ID not in entry.data or CONF_API_KEY not in entry.data:
+        LOGGER.info(
+            "Updating intellifire config entry for %s with api information",
+            entry.unique_id,
+        )
+        cloud_api = IntellifireAPICloud()
+        await cloud_api.login(
+            username=entry.data[CONF_USERNAME],
+            password=entry.data[CONF_PASSWORD],
+        )
+        api_key = cloud_api.get_fireplace_api_key()
+        user_id = cloud_api.get_user_id()
```

Comment:
Should this be stored in the `entry.data` to avoid fetching it again on next startup?

Reply:
It only drops into this loop if one of the users is migrating to this version ... its my migration w/out using a migration handler and having to bump the image.

If we don't have a USER_ID or API_KEY in the entry data they get added. If I did this wrong happy to change it :)

<img width="556" alt="image" src="https://user-images.githubusercontent.com/6491743/176455318-b490ffa5-874e-414a-a730-a4a118e4ad74.png">

Reply:
If the intent is to save the api key and user id in the config entry call `hass.config_entries.async_update_entry`